### ISSUE TITLE
Question about alternative HTTP clients

### ISSUE THREAD

**Issue by anxiousmodernman** [spacemonkeygo/openssl#52]
Based on the code comment here: https://github.com/spacemonkeygo/openssl/blob/master/http.go#L51

... it seems that an HTTP _client_ is not implemented. Are there any known/recommended HTTP clients that can accept/embed the connection object created by this library? I do not need all of the functionality of the standard lib, just need to make REST calls against a secure service.

**Comment by jtolds** [comment]
Yeah, so, that's a relatively stale comment back from Go 1.2, thanks for the reminder. I'm going to leave this issue open until I can actually fix the documentation, but just so you know, you should be able to set the DialTLS method in current Go stdlib http.Transport objects.

Something like this (haven't checked for validity) should work:

```
   client = &http.Client{Transport: &http.Transport{
      DialTLS: func(network, address string) (net.Conn, error) {
        return openssl.Dial(network, address, ctx, 0)
      }}}

One downside to this approach is that you end up not using all of the settings you get on the DefaultTransport such as timeouts, so you'll have to reset timeouts again on this http.Client yourself (see how net/http's DefaultTransport is initialized)
